### PR TITLE
Add Terraform module

### LINKED ISSUE #18: Implement Terraform module

### PR DESCRIPTION
**Pull request by zpatrick**
Closes #18

### CODE REVIEW

File: vendor/github.com/zpatrick/fireball/router.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ func (r *BasicRouter) matchRoute(route *Route, req *http.Request) *RouteMatch {
 		return nil
 	}
 
-	pathVariables, ok := r.matchPathVariables(route, req.URL.Path)
+	pathVariables, ok := r.matchPathVariables(route, req.URL.RawPath)
```
Comment (sesh-kebab) [inline vendor/github.com/zpatrick/fireball/router.go]: What happens in the event that fireball is updated and needs to be updated in d.ims.io?

File: terraform/main.tf [hunk 2]
```diff
@@ -0,0 +1,82 @@
+data "aws_caller_identity" "current" {}
+
+resource "aws_dynamodb_table" "dimsio" {
+  name           = "${var.dynamodb_table_name}"
+  read_capacity  = 1
```
Comment (sesh-kebab) [inline terraform/main.tf]: Is potential throttling a concern with determining read & write capacities? Would enabling auto-scaling throughput capacity be a feasible option to cater for a more hands-off approach if this service were to be hit with throttling issues?

File: terraform/main.tf [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+resource "aws_iam_user" "dimsio" {
+  name = "${var.iam_user_name}"
+}
+
+resource "aws_iam_access_key" "dimsio" {
+  user = "${aws_iam_user.dimsio.name}"
+}
+
+resource "aws_iam_user_policy" "dimsio" {
```
Comment (sesh-kebab) [inline terraform/main.tf]: 👍

File: vendor/github.com/zpatrick/fireball/router.go (hunk 1 above)
Comment (zpatrick) [inline vendor/github.com/zpatrick/fireball/router.go]: I think we'd treat it like a change set to d.ims.io's source code

File: terraform/main.tf (hunk 2 above)
Comment (zpatrick) [inline terraform/main.tf]: Yes, throttling is a potential concern, I think we should at least make this a terraform variable for now.  I'm hoping we can get away with finding the "right balance" and seeing if that works before jumping into autoscaling.

**Comment by zpatrick** [comment]
> For future changes, do you anticipate rolling out a new deploy live to prod or a blue/green kind of deployment for each incremental revision?

I haven't thought about that yet to be honest. Luckily, that really depends on who/what is consuming this module, and not really the module itself.

**Pull request merged**